cpp/src/graphar/high-level/graph_reader.h

```diff
@@ -172,40 +179,80 @@ class VertexIter {
    * @param offset The current offset of the readers.
    */
   explicit VertexIter(const std::shared_ptr<VertexInfo>& vertex_info,
-                      const std::string& prefix, IdType offset) noexcept {
+                      const std::string& prefix, IdType offset,
+                      const std::vector<std::string>& labels,
+                      const bool& is_filtered = false,
+                      const std::vector<IdType>& filtered_ids = {}) noexcept {
+    if (!labels.empty()) {
+      labels_ = labels;
+      label_reader_ =
+          VertexPropertyArrowChunkReader(vertex_info, labels, prefix);
+    }
     for (const auto& pg : vertex_info->GetPropertyGroups()) {
       readers_.emplace_back(vertex_info, pg, prefix);
     }
+    is_filtered_ = is_filtered;
```

Comment:
is_filtered_ is used for mark the collection/iter is or not a result from filtering